scikeras/wrappers.py

```diff
@@ -394,46 +386,55 @@ def _fit_keras_model(self, X, y, sample_weight, warm_start):
         else:
             hist = self.model_.fit(x=X, y=y, **fit_args)
 
-        if warm_start:
-            if not hasattr(self, "history_"):
-                self.history_ = defaultdict(list)
-            self.history_ = {
-                get_metric_full_name(k): self.history_[get_metric_full_name(k)]
-                + hist.history[k]
-                for k in hist.history.keys()
-            }
-        else:
-            self.history_ = hist.history
-        self.is_fitted_ = True
+        if not warm_start or not hasattr(self, "history_"):
+            self.history_ = defaultdict(list)
+
+        for key, val in hist.history.items():
+            try:
+                key = metric_name(key)
+            except ValueError as e:
+                # Keras puts keys like "val_accuracy" and "loss" and
+                # "val_loss" in hist.history.
+                if "Unknown metric function" not in str(e):
+                    raise e
+            self.history_[key] += [val]
 
         # return self to allow fit_transform and such to work
         return self
 
-    def _check_output_model_compatibility(self, y):
-        """Checks that the model output number and y shape match, reshape as needed.
+    def _check_model_compatibility(self, y: np.ndarray) -> None:
+        """Checks that the model output number and y shape match.
 
-        This is mainly in place to avoid cryptic TF errors.
+        This is in place to avoid cryptic TF errors.
         """
         # check if this is a multi-output model
         if self.model_n_outputs_ != len(self.model_.outputs):
```

Comment:
Should SciKeras perform a check on the number of inputs too?

Reply:
I guess we could, but I think that can be another PR, it doesn't have to happen in this method.

Reply:
Looks like #106 is going to track this. Resolving this thread.